Apps/GatewayApi/src/Services/RegistrationService.cs

```diff
@@ -121,14 +98,14 @@ public async Task<UserProfileModel> CreateUserProfileAsync(CreateUserRequest cre
             // queue verification email
             if (emailVerification != null && !isEmailVerified)
             {
-                await emailQueueService.QueueNewEmailAsync(emailVerification.Email, true, ct);
+                await jobService.SendEmailAsync(emailVerification.Email, true, ct);
             }
 
             // notify partners about new account
             await this.NotifyAccountCreated(profile, requestedEmail, requestedSmsNumber, isEmailVerified, smsVerification?.SmsValidationCode, ct);
 
             // build and return model
-            UserProfileModel userProfileModel = await userProfileService.GetUserProfileAsync(hdid, jwtAuthTime, ct);
+            UserProfileModel userProfileModel = await userProfileModelService.BuildUserProfileModelAsync(profile, this.userProfileHistoryRecordLimit, ct);
             logger.LogTrace("Finished creating user profile... {Hdid}", dbResult.Payload.HdId);
             return userProfileModel;
```

Comment:
The changes here work well to clean up the awkward logic after the previous refactor.

Reply:
No changes made.